SparkPost/2web2ui
# SA-2209 - (UI) Add missing unit tests for global Analytics Report components

[SA-2209](https://sparkpost.atlassian.net/browse/SA-2209)

### What Changed

- Adds unit tests for global Analytics Report components
- [Renames a click handler prop](https://github.com/SparkPost/2web2ui/pull/2091/commits/5548fc756fea22c5bc880bcc14eeb353fa541f76) for the sake of clarity
- Upgrades React Testing Library dependencies (thought it was worth doing):
    - `@testing-library/react`
    - `@testing-library/react-hooks`
    - `@testing-library/user-event`
    - `@testing-library/jest-dom`
- Updates an existing test that began to fail after updating `@testing-library/user-event`. Looks like some of the element attributes are set incorrectly - I filed [a follow-up ticket to address](https://sparkpost.atlassian.net/browse/FE-1386) this and other similar issues in the app related to the use of Downshift 

### How To Test

- Verify tests accurately test code paths within each component
- Verify tests pass in CI

### To Do
